pisa-proxy/runtime/mysql/src/server/server.rs

```diff
@@ -50,42 +54,109 @@ pub struct MySqlServer {
     // `concurrency_control_rule_idx` is index of concurrency_control rules
     // `concurrency_control_rule_idx` is required to add permits when the concurrency_control layer service is enabled
     concurrency_control_rule_idx: Option<usize>,
+}
 
-    pub metrics_collector: MySqlServerMetricsCollector,
-    // TODO: this should be a common property of proxy runtime
-    pub name: String,
+pub struct MySqlServerBuilder {
+    _name: String,
+    _socket: TcpStream,
+    _pcfg: ProxyConfig,
+    _buf: BytesMut,
+    _mysql_parser: Arc<Parser>,
+    _ast_cache: Arc<plMutex<ParserAstCache>>,
+    _is_quit: bool,
+    _concurrency_control_rule_idx: Option<usize>,
+    _metrics_collector: MySqlServerMetricsCollector,
+    _lb: Arc<Mutex<Box<dyn LoadBalance + Send + Sync>>>,
+    _pool: Pool<ClientConn>,
+    _plugin: Option<PluginPhase>,
 }
 
-impl MySqlServer {
-    pub async fn new(
-        client: TcpStream,
-        pool: Pool<ClientConn>,
-        lb: Arc<Mutex<BalanceType>>,
-        proxy_config: ProxyConfig,
-        parser: Arc<Parser>,
-        ast_cache: Arc<plMutex<ParserAstCache>>,
+impl MySqlServerBuilder {
+    pub fn new(
+        socket: TcpStream,
+        lb: Arc<Mutex<Box<dyn LoadBalance + Send + Sync>>>,
```

Comment:
lb has been changed to `enum`, update branch first

Reply:
fixed